fiftyone/operators/loader.py

```diff
@@ -87,13 +90,21 @@ def register_module(plugin_definition, mod):
     Returns:
         the :class:`PluginContext` for the plugin
     """
+    pctx = None
     try:
         pctx = PluginContext(plugin_definition)
-        KNOWN_PLUGIN_CONTEXTS[pctx.name] = pctx
         mod.register(pctx)
     except Exception as e:
-        errors = [traceback.format_exc()]
-        pctx.errors = errors
+        if pctx:
+            pctx.errors.append(traceback.format_exc())
+    finally:
+        if pctx and pctx.name:
+            KNOWN_PLUGIN_CONTEXTS[pctx.name] = pctx
+        else:
+            # If plugin context is None, then there is no plugin name to
```

Comment:
We should print the plugin_definition.directory

Reply:
added logging